core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -168,6 +173,37 @@ class BlockManagerMasterEndpoint(
       stop()
   }
 
+  private def handleFailure[T](
+      blockType: String,
+      blockId: String,
+      bmId: BlockManagerId,
+      defaultValue: T): PartialFunction[Throwable, T] = {
```

Comment:
[nit] since this method deals with handling block removal failures explicitly, can we rename it to reflect the same. maybe something like: handleBlockRemovalFailure?

Reply:
Good idea!